File: exercise/c#/Day20/Day20.Tests/DiceBuilder.cs
namespace Day20.Tests
{
    public class DiceBuilder
    {
        private readonly int[] _dice;

        private DiceBuilder(int[] dice) => _dice = dice;

        public static DiceBuilder NewRoll(int dice1, int dice2, int dice3, int dice4, int dice5)
            => new([dice1, dice2, dice3, dice4, dice5]);

        public int[] Build() => _dice;

        public override string ToString() => $"[{string.Join(", ", _dice)}]";
    }
}

File: solution/ts/day20/tests/diceBuilder.ts
export class DiceBuilder {
    private readonly dice: number[];

    constructor(dice: number[]) {
        this.dice = dice;
    }

    static newRoll = (dice1: number, dice2: number, dice3: number, dice4: number, dice5: number): DiceBuilder =>
        new DiceBuilder([dice1, dice2, dice3, dice4, dice5]);

    build = (): number[] => this.dice;

    toString = () => this.dice.toString();
}